src/frameworks/foundation/ns_object.rs

```diff
@@ -16,10 +16,34 @@
 
 use super::ns_string::to_rust_string;
 use super::NSUInteger;
+use crate::dyld::{export_c_func, FunctionExports};
 use crate::mem::MutVoidPtr;
 use crate::objc::{
-    id, msg, msg_class, msg_send, objc_classes, Class, ClassExports, ObjC, TrivialHostObject,
+    id, msg, msg_class, msg_send, objc_classes, Class, ClassExports, ObjC, TrivialHostObject, SEL,
 };
+use crate::Environment;
+
+pub(super) fn NSClassFromString(
+    env: &mut Environment,
+    _class_name: id, // NSString
+) -> Class {
+    // TODO: actually implement this
+    println!("Warning: NSClassFromString not implemented, app will crash.");
+    return msg_class![env; NSObject alloc];
+}
```

Comment:
I won't merge these as-is because I think they're a bit too hacky and we should just implement them for real. For reference:

* `NSClassFromString` can use `env.objc.get_known_class`. Note that will panic if it doesn't find a class, but that's probably best for the time being.
* `NSLog` can share code with `printf`, though there are some small differences in how format strings work for the two.
* `respondsToSelector:` can actually check if the selector is supported. Some changes in `src/objc/` might be needed. We'll probably want to have some kind of warning if it returns `false` when used on a host class, though, as it's probably the app checking for functionality that we haven't implemented yet.

But to be clear, I don't expect you to implement these unless you want to. I'm sure they're common enough that I'll get round to them eventually.

Reply:
Actually, I am probably going to implement `NSLog` today, so don't worry about that one. :)

Reply:
I'm probably going to leave this as-is for now.

Ideally, NSClassFromString should return `nil` if the class does not exist, because some apps use NSClassFromString when checking if an API exists or not. In fact, even Apple did this in early sample code for iPhone OS (see EAGLView.m): https://developer.apple.com/library/archive/samplecode/GLES2Sample/Introduction/Intro.html#//apple_ref/doc/uid/DTS40009188-Intro-DontLinkElementID_2

Reply:
Yeah, I guess the real implementation of that would need to be full of special cases.